### PR TITLE
Add sticky posts support and styles

### LINKED ISSUE #8: Pinned post support
Content editors should be able to pin a blog post using the WordPress editor

In WordPress this feature is called "Sticky"

I've gone ahead and made the post with slug `i-love-you-buddy` sticky. If you look in the JSON response, the post that's sticky has an attribute of `sticky: true`.

http://54.243.240.9/wp-json/wp/v2/posts

You can make multiple posts sticky.

We probably will want to find the posts that are sticky within the posts array, pull them out, and append them to the top of the posts list when rendering them to the page

We should additionally add some kind of icon to identify a sticky post. Maybe a pin icon? I'm open to suggestions. The project includes `react-icons` and currently makes use of Font Awesome. We should use an icon that matches that style.

## Questions
* Should the "sticky" posts show twice? Once at the top and once in it's natural location? How does Twitter handle pinned Tweets? What does a typical WP site do?

### PR DESCRIPTION
**Pull request by GuilleAngulo**
**IMPORTANT**: Waiting [this release](https://github.com/wp-graphql/wp-graphql/pull/1513) to be available

Fixes #8

### Description
It adds to the GraphQL queries the boolean field _isSticky_. With this new information, is implemented a conditional style at _PostCard_, in order to apply the new style in case of a sticky post. Also the _posts_ array is sorted before iteration to put sticky posts at the top of the list. It is added this attribute to the post _metadata_ in order to draw a pin at the post detail page.

### Review

- [ ] Released support to **_isSticky_** field at [wp-graphql](https://github.com/wp-graphql/wp-graphql) plugin

### Screenshots

|Index Page                    |Post Detail
| ---------------------------------- | ---------------------------------- |
| sticky-index |  sticky-post|

### CODE REVIEW

File: src/components/PostCard/PostCard.js [hunk 1]
```diff
@@ -26,7 +27,8 @@ const PostCard = ({ post, options = {} }) => {
   }
 
   return (
-    <div className={styles.postCard}>
+    <div className={isSticky ? `${styles.postCard} ${styles.postCardSticky}` : `${styles.postCard}`}>
```
Comment (colbyfayock) [inline src/components/PostCard/PostCard.js:30]: Another thing we can consider doing is pulling this up into a variable, might make it easier to read, but i don't feel strongly about it

for instance, could do something like:

```
let className = styles.postCard;

if ( isSticky ) {
  className = `${className} ${styles.postCardSticky}`
}
```

**Comment by colbyfayock** [comment]
the code looks great @GuilleAngulo! excited to poke at this and get it merged in once the change is on the main branch :)

File: src/components/PostCard/PostCard.js (hunk 1 above)
Comment (GuilleAngulo) [reply src/components/PostCard/PostCard.js:30]: Nice! I agree, will make the changes 👍

**Comment by colbyfayock** [comment]
hey @GuilleAngulo just updated the WP instance to use 0.14.0 and it works great!

Can you fix the merge conflicts and then we can get this in?

image

**Comment by GuilleAngulo** [comment]
Sure thing! 
One quick note: I was cheking it these days and realised that exists one conflict when sorting the array at the component. It happened because the array was sorted in the pre-render and then at client hydration sorted again, so existed a conflict between server and client (because of the way of sorting ` post.isSticky ? -1 : 1`. For example for three sticky posts the order of them at the top will difffer for the two sortings, because the last sticky post will become first.

So my idea was to move this sorting function to the getStaticProps(), and this way the sorting one time only when at pre-render:

```
export async function getStaticProps() {
  const { posts } = await getAllPosts();
  return {
    props: {
      posts: posts.sort((post) => (post.isSticky ? -1 : 1)),
    },
  };
}
```

**Comment by colbyfayock** [comment]
cool good catch, i think that works well

**Pull request merged**